### PR TITLE
🔡 Update default core Gradio font 

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR updates our current Sans font. I've opted to stick with Google Fonts - and found a couple that I think work well with our current monospace font, `IBM Plex Mono` which I already like (though I do also like [Reddit Mono](https://fonts.google.com/specimen/Reddit+Mono)). 

This PR implements the `Asap` font, but I also mentioned in Slack that I also like `IBM Plex Sans`, which naturally pairs well with its `Mono` alternative. 

| Asap | 
|--------|
| Screenshot 2024-08-28 at 13 16 51 | 
| Screenshot 2024-08-28 at 13 17 23 |
| Screenshot 2024-08-28 at 13 19 30 |
| Website text (if we do change it) 
image |

---

| IBM Plex Sans |
|--------|
| image |
| image |
| Screenshot 2024-08-28 at 13 21 48 |
| Screenshot 2024-08-28 at 13 21 50 | 

Closes: #1224 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: guides/09_other-tutorials/theming-guide.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ You could also create your own custom `Size` objects and pass them in.
 
 The final 2 constructor arguments set the fonts of the theme. You can pass a list of fonts to each of these arguments to specify fallbacks. If you provide a string, it will be loaded as a system font. If you provide a `gradio.themes.GoogleFont`, the font will be loaded from Google Fonts.
 
-- `font`: This sets the primary font of the theme. In the default theme, this is set to `gradio.themes.GoogleFont("Source Sans Pro")`.
+- `font`: This sets the primary font of the theme. In the default theme, this is set to `gradio.themes.GoogleFont("Asap")`.
```
Comment (abidlabs) [inline guides/09_other-tutorials/theming-guide.md:159]: Needs to be updated

File: js/_cdn-test/index.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
 			html {
 				/* background: #111; */
 				/* color: #eee; */
-				font-family: Source Sans Pro;
+				font-family: Asap;
```
Comment (abidlabs) [inline js/_cdn-test/index.html:8]: Needs to be updated I think?

**Review by abidlabs — APPROVED** [review]
LGTM! Left a few minor comments

**Comment by dawoodkhan82** [comment]
I personally like Asap better than IBM Plex Sans. But fine with either!

**Review by dawoodkhan82 — APPROVED** [review]

**Pull request merged**